packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,269 @@
+import Popup from '../Popup/Popup';
+import {
+  IRestrictedDatesOptions,
+  IDay,
+  DayOfWeek,
+  FirstWeekOfYear,
+  DateRangeType,
+  IDateGridStrings,
+  formatMonthDayYear,
+  IDayGridOptions,
+} from '@fluentui/date-time-utilities';
+import { default as DatepickerCalendar, DatepickerCalendarProps } from './DatepickerCalendar';
+
+const DEFAULT_STRINGS: IDateGridStrings = {
+  months: [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ],
+  shortMonths: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+  days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+  shortDays: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
+};
+
+export interface IDateFormatting {
+  /**
+   * Format the date according to specified function.
+   * Intended use case is localization.
+   */
+  format?: (date: Date) => string;
+
+  /**
+   * Parse date from string representation into Date type.
+   */
+  parse?: (date: string) => Date;
+}
+
+export interface IDatepickerOptions extends IRestrictedDatesOptions {
+  /**
+   * The first day of the week for your locale.
+   */
+  firstDayOfWeek?: DayOfWeek;
+
+  /**
+   * Defines when the first week of the year should start, FirstWeekOfYear.FirstDay,
+   * FirstWeekOfYear.FirstFullWeek or FirstWeekOfYear.FirstFourDayWeek are the possible values
+   */
+  firstWeekOfYear?: FirstWeekOfYear;
+
+  /**
+   * The date range type indicating how  many days should be selected as the user
+   * selects days
+   */
+  dateRangeType?: DateRangeType;
+
+  /**
+   * The number of days to select while dateRangeType === DateRangeType.Day. Used in order to have multi-day
+   * views.
+   */
+  daysToSelectInDayView?: number;
+
+  /**
+   * Value of today. If null, current time in client machine will be used.
+   */
+  today?: Date;
+
+  /**
+   * Whether the calendar should show the week number (weeks 1 to 53) before each week row
+   */
+  showWeekNumbers?: boolean;
+
+  /**
+   * The days that are selectable when `dateRangeType` is WorkWeek.
+   * If `dateRangeType` is not WorkWeek this property does nothing.
+   */
+  workWeekDays?: DayOfWeek[];
+}
+
+export interface DatepickerProps extends IDatepickerOptions, IDateFormatting, UIComponentProps {
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility<DatepickerBehaviorProps>;
+
+  /** Datepicker shows it is currently unable to be interacted with. */
+  disabled?: boolean;
+
+  /** Datepicker shows it is currently unable to be interacted with. */
+  isRequired?: boolean;
+
+  /**
+   * Called on change of the date.
+   *
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props and proposed value.
+   */
+  onDayChange?: (e: React.MouseEvent<HTMLElement>, data: DatepickerProps & { value: IDay }) => void;
```

Comment:
We usually use `ComponentEventHandler<DatepickerProps>` for the callbacks definition.

Also, isn't value part of the props already?

Reply:
The selected value is a state in this case, which is why I create a composed type. 

We had a talk with @layershifter  that we should actually switch away from using the ComponentEventHandler as it internally relies on SyntheticEvent which leaves you with too broad of a type definition.

Reply:
I understand, but would be nice to do this change in a batch for all components. It is confusing why this typing is different than all the others

Reply:
Good point. After further discussion with shift, I changed the type back to ComponentEventHandler.

Reply:
Created the following task:
Task [1041728](https://dev.azure.com/domoreexp/MSTeams/_workitems/edit/945977): Move away from using ComponentEventHandler